src/core/utils.js

```diff
@@ -104,41 +92,54 @@ function markAsOffending(elem, msg, title) {
 }
 
 // STRING HELPERS
+/**
+ * @param {"conjunction"|"disjunction"} type
+ * @param {"long"|"narrow"} style
+ */
+function joinFactory(type, style = "long") {
+  const formatter = new Intl.ListFormat(docLang, { style, type });
+  /**
+   * @param {string[]} items
+   * @param {(value: string, index: number, array: string[]) => any} [mapper]
+   */
+  return (items, mapper) => {
+    let elemCount = 0;
+    if (!mapper) {
+      return [formatter.format(items)];
+    }
+    return formatter.formatToParts(items).map(({ type, value }) => {
+      if (type === "element") {
+        return mapper(value, elemCount++, items);
```

Comment:
Looks like we don't use it.
```suggestion
        return mapper(value, elemCount++);
```

Reply:
I'd rather sent the interface compliant map.

Reply:
(even though we don't use it)